package.json

```
@@ -19,7 +19,8 @@
     "test": "npm run lint && jest",
     "lint": "npm run lint:js && npm run lint:scss",
     "lint:js": "eslint --cache --cache-location .cache/eslint --cache-strategy content --color --ignore-path .gitignore \"**/*.{cjs,js,mjs}\"",
-    "lint:scss": "stylelint \"**/*.scss\""
+    "lint:scss": "stylelint \"**/*.scss\"",
+    "validate": "npm run build && html-validate \"deploy/public/**/index.html\""
```

Shall we make this `lint:html` and get it running on GitHub Actions? 🙌 

```suggestion
    "lint:html": "html-validate \"deploy/public/**/index.html\""
```

Or alternatively could set up an npm "post" script?

```suggestion
    "postbuild": "html-validate \"deploy/public/**/index.html\""
```

---

Hmm, neither seems fully satisfying to me - having it in Github Actions means we don't catch things locally, but setting up a `postbuild` script means it won't run on `npm run test`.

And adding it to `test` means we'll have to include the 8-10 second build step for that script, which is also annoying.

---

Ahh, but it would still be lovely to see all your hard work in GitHub Actions

Could always call into the task from multiple places where needed?

```json
"pretest": "npm run validate"
"postbuild": "npm run validate"
```

Possibly a daft example 🙈

---

For now, I've just made the `lint:html` script. I briefly had it running in Github actions, but it was having parsing errors which I'll try and figure out.

---

@domoscargin Would you mind if we resolved the GitHub Actions checks before merging?

Would be great to validate all my Markdown and code block changes automatically

---

Yeah, sorry, I should've popped this back to draft maybe - something breaks the Github Actions and I haven't had time to dig into what it is. Some kind of parsing maybe.

---

https://github.com/alphagov/govuk-design-system/actions/runs/4222561918/jobs/7331207893#step:7:28

---

(Got some time set aside to have a look at this later, but if it's an obvious fix...)

---

Ah it was an invalid config at our end which hit a bug in `html-validate`

They're using AJV for JSON schema validation, but _maaybe_ it's not working with our JS config?

If you log the config validation error before it's formatted you get:

>SchemaValidationError [UserError]: Rule configuration error: /rules/form-dup-name/1/shared/3 must be equal to one of the allowed values: radio, checkbox, submit

---

Could there be an outdated version of html-validate cached?

---

Ah, no! My bad - I haven't updated my package.json.

---

Yup, just needed to bump `html-validate` to get the updated rule and it's all working now. Thanks for the helpful error log wrangling!